orchestra/communication/staffing.py

```diff
@@ -117,19 +120,58 @@ def send_staffing_requests(
                 Q(last_inquiry_sent__lte=cutoff_datetime)))
 
     for request in requests:
-        send_request_inquiries(staffbot, request, worker_batch_size)
+        send_request_inquiries(staffbot, request)
+
+
+def get_worker_priorities(workers_with_inquiries):
+    def get_workers_in_priority_order():
+        return (Worker.objects
+                .exclude(id__in=workers_with_inquiries)
+                .order_by('-staffing_priority', '?'))
+
+    # Ping 2 workers in priority order every 2 minutes
+    return [(get_workers_in_priority_order, 2, 2)]
+
 
+def _get_worker_priority_functions(request, workers_with_inquiries):
+    workflow_name = request.task.project.workflow_version.name
 
-def send_request_inquiries(staffbot, request, worker_batch_size):
-    # Get Workers that haven't already received an inquiry.
+    if (hasattr(settings, 'ORCHESTRA_WORKER_PRIORITY_SORT_PATHS') and
+            workflow_name in settings.ORCHESTRA_WORKER_PRIORITY_SORT_PATHS):
+        try:
+            worker_sort_function_path = (
+                settings.ORCHESTRA_WORKER_PRIORITY_SORT_PATHS[workflow_name]
+            )
+            worker_sort_function = locate(worker_sort_function_path)
+            return worker_sort_function(request,
+                                        workers_with_inquiries)
+        except Exception as e:
+            raise WorkerSortError(e)
+    else:
+        return get_worker_priorities(workers_with_inquiries)
+
+
+def send_request_inquiries(staffbot, request):
     workers_with_inquiries = (StaffingRequestInquiry.objects.filter(
         request=request).distinct().values_list(
             'communication_preference__worker__id', flat=True))
-    # Sort Workers by their staffing priority first, and then randomly
-    # within competing staffing priorities.
-    workers = (Worker.objects
-               .exclude(id__in=workers_with_inquiries)
-               .order_by('-staffing_priority', '?'))
+
+    worker_priorities = _get_worker_priority_functions(request,
+                                                       workers_with_inquiries)
+    workers = None
+    stage = 0
+    batch_size = 0
+    interval = 0
+
+    while workers is None and stage < len(worker_priorities):
+        get_workers_func, batch_size, interval = worker_priorities[stage](
+            workers_with_inquiries)
+        cutoff_datetime = timezone.now() - timedelta(interval)
```

Comment:
i don't get the cutoff logic---can you explain? i suspect we don't want it, but maybe i misunderstand it :)